incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -373,7 +377,10 @@ func (r *HierarchyConfigReconciler) syncParent(log logr.Logger, inst *api.Hierar
 
 	// Sync this namespace with its current parent.
 	curParent := r.Forest.Get(inst.Spec.Parent)
-	if curParent != nil && !curParent.Exists() {
+	if config.ExcludedNamespaces[inst.Spec.Parent] {
+		log.Info("The parent is an excluded namespace that should not be set as a parent", "parent", inst.Spec.Parent)
+		ns.SetCondition(api.ConditionBadConfiguration, api.ReasonIllegalParent, fmt.Sprintf("Parent %q is an excluded namespace. HNC will treat this namespace as if it has no parent.", inst.Spec.Parent))
```

Comment:
Why is this BadConfiguration and not ActivitiesHalted?

Reply:
I was initially thinking the activities can still continue as if this namespace has no parent, but since `ParentMissing` condition is similar to this case and it didn't treat the namespace as if it doesn't have a parent, I guess it makes sense to have this one as `ActivitiesHalted` too. In that case, the descendants would have `ActivitiesHalted: AncestorHaltActivities` condition too, so that it's more explicit to unveil the issue to users, SGTY?